src/components/task-plan/footer.cjsx

```diff
@@ -12,8 +12,14 @@ PlanFooter = React.createClass
     courseId: React.PropTypes.any.isRequired
     clickedSelectProblem: React.PropTypes.func
 
+  saved: ->
+    courseId = @props.courseId
+    TaskPlanStore.removeChangeListener(@saved)
```

Comment:
What if the `save` errors instead of calling `Actions.saved`? I think you might need to test the Action being called. I remember suggesting something similar (like `Actions.saved.on(@saved)`) and then detaching `@saved` on `componentDidUnmount` to @pandafulmanda in another part of the code.